src/multipeek_impl.rs

```diff
@@ -38,6 +38,7 @@ impl<I: Iterator> MultiPeek<I> {
     /// Works exactly like `.next()` with the only difference that it doesn't
     /// advance itself. `.peek()` can be called multiple times, to peek
     /// further ahead.
+    /// When `.next()` is called, reset the peeking “cursor”.
```

Comment:
I like the idea of documenting this. However, when I read this, it was not immediately obvious to me what the "cursor" even is. Maybe we could use examples in there?